src/patterns/ec2-app/base.ts

```diff
@@ -307,6 +340,15 @@ export class GuEc2App {
       withoutImdsv2,
     } = props;
 
+    // We should really prevent users from doing this via the type system,
```

Comment:
We have quite a few props related to logging now:

`accessLogging`
`applicationLogging`
`roleConfiguration.withoutLogShipping`

We are also configuring the tag for `cloud-init-output.log` shipping automatically: https://github.com/guardian/cdk/pull/1211.

I was tempted to change the API to introduce a single `loggingConfiguration` interface, which explained all of this and tried to make the configuration a bit more consistent. However, we have been making a lot of breaking changes recently and this feels like something that can wait.